src/rdkafka_admin.c

```diff
@@ -2737,3 +2964,474 @@ rd_kafka_DescribeConfigs_result_resources (
 }
 
 /**@}*/
+
+/**
+ * @name Delete Records
+ * @{
+ *
+ *
+ *
+ *
+ */
+
+static void rd_kafka_topic_partition_list_free (void *ptr) {
```

Comment:
There's a rd_kafka_topic_partition_list_destroy_free() already that you can use.